src/account.c

```diff
@@ -81,29 +81,40 @@ static int param_u32(uint32_t *v, const struct pl *params, const char *name)
 static int stunsrv_decode(struct account *acc, const struct sip_addr *aor)
 {
 	struct pl srv, tmp;
+	struct uri uri;
 	int err;
 
 	if (!acc || !aor)
 		return EINVAL;
 
+	memset(&uri, 0, sizeof(uri));
 	if (0 == msg_param_decode(&aor->params, "stunserver", &srv)) {
 
 		info("using stunserver: '%r'\n", &srv);
 
-		err = stunuri_decode(&acc->stun_host, &srv);
+		err = uri_decode(&uri, &srv);
 		if (err) {
-			warning("account: decode '%r' failed: %m\n",
+			warning("account: decode '%r' failed (%m)\n",
 				&srv, err);
+			return err;
 		}
+
+		stunuri_decode_uri(&acc->stun_host, &uri);
 	}
 
 	err = 0;
 
 	if (0 == msg_param_exists(&aor->params, "stunuser", &tmp))
 		err |= param_dstr(&acc->stun_user, &aor->params, "stunuser");
+	else if (pl_isset(&uri.user))
+		err |= re_sdprintf(&acc->stun_user, "%H",
+					uri_user_unescape, &uri.user);
 
 	if (0 == msg_param_exists(&aor->params, "stunpass", &tmp))
 		err |= param_dstr(&acc->stun_pass, &aor->params, "stunpass");
+	else if (pl_isset(&uri.password))
```

Comment:
we can print a message here saying "deprecated field used"

Reply:
[d7b8d0d](https://github.com/baresip/baresip/pull/1960/commits/d7b8d0d951f7800f5cfcfee67bf026b922d48149)
It seems to me that the most correct place to generate a message is inside the stunuri_decode_uri method. In the future, if the password in uri is completely disabled, it is possible to return an error